### PR TITLE
Dev votes user

### PR DESCRIPTION
**Pull request by arkanoider**
Hi @grunch 

this is really a draft, not compiling a still missing things. The idea is to send notification to buyer/seller after a successful trade is done. Here the line:

https://github.com/arkanoider/mostro/blob/699c4999c8c977edf80a10ecb99398413a694b46/src/app/release.rs#L108

Here the messages:
https://github.com/arkanoider/mostro/blob/699c4999c8c977edf80a10ecb99398413a694b46/src/util.rs#L331

I send the order as content so client can get the info they need and a new action vote.

When buyer/seller receive it they will send a new command vote with:

- Id order
- Npub of counterpart
- Vote ( 1-5 )

Mostro will receive a new action vote and start updating process:
- Make a request to relay of replacable events ( This lead to the issue to manage a not supporting NIP33 relay)
- Get the data of the user
- Calculate the new values 
- Update event

In my mind this is the starting point for the Action::Vote when mostro receive the message:

https://github.com/arkanoider/mostro/blob/699c4999c8c977edf80a10ecb99398413a694b46/src/app/vote_user.rs#L125

Still a bit messy...but you guys can give a look and tell me if you had any doubts!

Thanks for your patience!

### CODE REVIEW

**Comment by arkanoider** [comment]
Hi @grunch 

i did some some work and I think via have a minimal functional starting point.

I added a vote command on CLI side and the needed struct in mostro-core.

Basically now after a successful trade bueyer and seller can vote each other the votes are marked in the db so they can just give one vote for a specific order id and counterpart.

To compile in a clean way we need to update mostro core and maybe bump version and as I told you I add a github dep in cargo.toml to avoid an error caused by bitcoin-hashes lib ( thanks to @yukibtc ).

Another point in my mind is that it could be nice to know if the relay we are working supports nip33...but maybe just getting the newer event is enogh.

Waiting for your comments and ready to help for merging and compiling.

**Review by grunch — CHANGES_REQUESTED** [review]

File: src/db.rs [hunk 1]
```diff
@@ -221,6 +221,32 @@ pub async fn update_order_event_id_status(
     Ok(rows_affected > 0)
 }
 
+pub async fn update_order_event_id_vote_status(
```
Comment (grunch) [inline src/db.rs:224]: and this for `update_order_event_id_rate_status`

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
+{
```
[the remark this answers is not in the document]
Comment (arkanoider) [reply .vscode/launch.json:1]: Sorry it went in the commit...it's just the debugger config...throw away!

**Comment by grunch** [comment]
When we add changes to database and/or add new functions that interact with the database we have to run `cargo sqlx prepare` which update `sqlx-data.json`, this is needed to pass github tests

File: src/db.rs (hunk 1 above)
Comment (arkanoider) [reply src/db.rs:224]: yep!

File: Cargo.toml [hunk 3]
```diff
@@ -38,4 +39,4 @@ reqwest = { version = "0.11", features = ["json"] }
 mostro-core = "0.1.8"
```
[the remark this answers is not in the document]
Comment (arkanoider) [reply Cargo.toml:39]: > When we add changes to database and/or add new functions that interact with the database we have to run `cargo sqlx prepare` which update `sqlx-data.json`, this is needed to pass github tests

We have to definitively add to yaml file to automate the process..

File: .vscode/launch.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (arkanoider) [reply .vscode/launch.json:1]: I will fix now...ten minutes and i push the new commit

**Review by grunch — APPROVED** [review]
LGTM

**Pull request merged**